### PR TITLE
raftstore: check store id before sending wakeup message

### LINKED ISSUE #12478: v5.4.1: after restart all tikv/pd/tidb, one tikv continue report error "invalid store ID 0, not found"
### What version of TiKV are you using?

./tikv-server -V
 TiKV
Release Version:   5.4.0
Edition:           Community
Git Commit Hash:   44d2e7e8d86566b4b270c1ae39a4956e0f74a6e8
Git Commit Branch: heads/refs/tags/v5.4.1
UTC Build Time:    2022-04-29 06:35:41
Rust Version:      rustc 1.56.0-nightly (2faabf579 2021-07-27)
Enable Features:   jemalloc mem-profiling portable sse test-engines-rocksdb cloud-aws cloud-gcp cloud-azure
Profile:           dist_release

### Steps to reproduce

tcms test plan:
[endless-range-partition-oltp-sysbench-oltp-read-write@v5.4.1]

following tidb global config has been set.

tidb_partition_prune_mode='dynamic' ,  
tidb_replica_read='follower'

### What did you expect?
tikv should works fine.

### What did happened?
Tikv heartbeat with pd disconnect after that, no region message update from pd monitor.

tikv logs:
Status { code: 2-UNKNOWN, message: \"invalid store ID 0, not found\", details: [] }))"]
tikv.log:[2022/05/10 02:28:24.557 +00:00] [ERROR] [[util.rs:460](http://util.rs:460/)] ["request failed"] [err_code=KV:PD:gRPC] [err="Grpc(RpcFailure(RpcStatus { code: 2-UNKNOWN, message: \"invalid store ID 0, not found\", details: [] }))"]
tikv.log:[2022/05/10 02:28:25.558 +00:00] [ERROR] [[util.rs:460](http://util.rs:460/)] ["request failed"] [err_code=KV:PD:gRPC] [err="Grpc(RpcFailure(RpcStatus { code: 2-UNKNOWN, message: \"invalid store ID 0, not found\", details: [] }))"]
tikv.log:[2022/05/10 02:28:25.565 +00:00] [ERROR] [[util.rs:460](http://util.rs:460/)] ["request failed"] [err_code=KV:PD:gRPC] [err="Grpc(RpcFailure(RpcStatus { code: 2-UNKNOWN, message: \"invalid store ID 0, not found\", details: [] }))"]
tikv.log:[2022/05/10 02:28:25.565 +00:00] [ERROR] [[raft_client.rs:749](http://raft_client.rs:749/)] ["resolve store address failed"] [err_code=KV:Unknown] [err="Other(\"[src/server/[resolve.rs:102](http://resolve.rs:102/)]: RpcFailure: 2-UNKNOWN invalid store ID 0, not found\")"] [msg="region_id: 134700 from_peer { id: 134703 store_id: 6 } to_peer {} region_epoch { conf_ver: 11 version: 1138 } extra_msg {}"] [store_id=0]
tikv.log:[2022/05/10 02:28:25.565 +00:00] [ERROR] [[util.rs:460](http://util.rs:460/)] ["request failed"] [err_code=KV:PD:gRPC] [err="Grpc(RpcFailure(RpcStatus { code: 2-UNKNOWN, message: \"invalid store ID 0, not found\", details: [] }))"]

### PR DESCRIPTION
**Pull request by 5kbpers**
Signed-off-by: 5kbpers <hustmh@gmail.com>

### What is changed and how it works?

Issue Number: Close #12478

What's Changed:

```commit-message
check store id before sending wakeup message
```

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test
- Integration test

### Release note 

```release-note
Fix a bug that sometimes generates a message with zero store id when doing follower read
```

### CODE REVIEW

File: components/raftstore/src/store/worker/pd.rs [hunk 1]
```diff
@@ -1609,9 +1609,11 @@ where
         let f = async move {
             match resp.await {
                 Ok(Some((region, leader))) => {
-                    let msg = CasualMessage::QueryRegionLeaderResp { region, leader };
-                    if let Err(e) = router.send(region_id, PeerMsg::CasualMessage(msg)) {
-                        error!("send region info message failed"; "region_id" => region_id, "err" => ?e);
+                    if leader.get_store_id() != 0 {
```
Comment (tonyxuqqi) [inline components/raftstore/src/store/worker/pd.rs:1612]: when does get_store_id() = 0?

**Review by gengliqi — APPROVED** [review]

File: components/raftstore/src/store/worker/pd.rs (hunk 1 above)
Comment (tonyxuqqi) [reply components/raftstore/src/store/worker/pd.rs:1612]: got it. The PD may not have the leader information.

File: components/test_raftstore/src/pd.rs [hunk 2]
```diff
@@ -1433,7 +1439,11 @@ impl PdClient for TestPdClient {
         let cluster = self.cluster.rl();
         match cluster.get_region_by_id(region_id) {
             Ok(resp) => {
-                let leader = cluster.leaders.get(&region_id).cloned().unwrap_or_default();
+                let leader = if self.trigger_leader_info_loss.load(Ordering::SeqCst) {
```
Comment (tonyxuqqi) [inline components/test_raftstore/src/pd.rs:1442]: question: in real world, how does PD have the region information, but does not have leader information?

**Review by tonyxuqqi — APPROVED** [review]

File: components/test_raftstore/src/pd.rs (hunk 2 above)
Comment (5kbpers) [reply components/test_raftstore/src/pd.rs:1442]: Leader information is not be persisted.

**Review by Connor1996 — APPROVED** [review]

**Pull request merged**